scripts/secrets_manager

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+
+function check_dependencies() {
+  echo '----------------------------'
+  echo 'Checking dependencies'
+  echo '----------------------------'
+
+  if ! [ -x "$(command -v age)" ]; then
+  	echo 'age is not installed. installing with `brew install age`' >&2
+  	brew install age;
+
+    if ! [ -x "$(command -v age)" ]; then
+  	  echo 'age is still not installed. installing with `brew install age`' >&2
+  	  exit 1
+	  else 
+      echo 'all dependencies installed' 
+    fi
+	else 
+    echo 'all dependencies installed' 
+	fi
+
+}
+
+function check_dependencies_silent() {
+  # Only emits messages upon failure & exit
+  if ! [ -x "$(command -v age)" ]; then
+  	brew install age;
+
+    if ! [ -x "$(command -v age)" ]; then
+  	  echo 'age is still not installed. installing with `brew install age`' >&2
+  	  exit 1
+	  else 
+      echo 'dependencies installed, re-run script' 
+      exit
+    fi
+	fi
+}
+
+function project_directory() {
+  script_path=$(readlink -f "${BASH_SOURCE[0]}")
+  script_dir=$(dirname "$script_path")
+  project_dir=$(dirname "$script_dir")
+  echo "$project_dir"
+}
+
+function check_for_existing_keys() {
+  set +e
```

Comment:
One problem I ran into was that I had generate the keys, but reset my codebase... So the script found the keys and exited without adding my public key to the file to be committed...  The workaround was that I had to manually delete the keys from my keychain.  A more robust check would be to ALSO check that the found public key exists in the file on the filesystem, and if it's not there, add it.  That would fix the above issue.

Reply:
good idea, will implement this suggestion

Reply:
added